### PR TITLE
Share assets from mobile to other apps

### PR DESCRIPTION
**Pull request by matthinc**
[#241](https://github.com/immich-app/immich/issues/241)

### CODE REVIEW

**Comment by matthinc** [comment]
It would be great if you could test the functionality on ios. It might behave differently than the native android sharing dialog.

File: mobile/lib/modules/asset_viewer/providers/image_viewer_page_state.provider.dart [hunk 1]
```diff
@@ -42,9 +46,23 @@ class ImageViewerStateNotifier extends StateNotifier<ImageViewerPageState> {
...
     state = state.copyWith(downloadAssetStatus: DownloadAssetStatus.idle);
   }
+
+  void shareAsset(AssetResponseDto asset, BuildContext context) async {
+    showDialog(
+      context: context,
+      builder: (BuildContext buildContext) {
+        _shareService
+            .shareAsset(asset)
+            .then((_) => Navigator.pop(buildContext));
```
Comment (alextran1502) [inline mobile/lib/modules/asset_viewer/providers/image_viewer_page_state.provider.dart:56]: Can we use `AutoRouter.of(context).pop()` to keep navigation within the `auto_router` package?

https://pub.dev/packages/auto_route

Comment (matthinc) [reply mobile/lib/modules/asset_viewer/providers/image_viewer_page_state.provider.dart]: This does not seem to work. DeleteDialog also uses `Navigator.of(buildContext).pop()`.

Comment (alextran1502) [reply mobile/lib/modules/asset_viewer/providers/image_viewer_page_state.provider.dart]: Ah, ok then. We can leave it as it

**Pull request merged**